File: backend/spotify-server/Models/Folder/Folder.cs
namespace spotify_server.Models.Folder;

public class Folder : IDefaultItem
{
    public string type {get;set;}
    public long id {get;set;}
    public long[] owner {get;set;}
    public string title {get;set;}
    public string[] contentsType {get;set;}
    public long[] contents {get;set;}
    public DateTime date {get;set;}
}

File: src/models/Items/Folder.ts
import Album from "./Album";
import ID from "../ID";
import ItemType from "../ItemType";
import Playlist, { PlaylistID } from "./Playlist";
import { ArtistID, ArtistView } from "./Artist";
import DefaultItem from "../DefaultItem";

export type FolderContent = Folder | Playlist;

interface Folder extends DefaultItem {
    type: ItemType.FOLDER;
    id: FolderID;
    title: string;
    owner: ArtistView[];
    contents: FolderContent[];
}

export type FolderID = ID;

export default Folder;